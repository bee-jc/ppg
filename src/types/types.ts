export interface Episode {
  id: number;
  name: string;
  image: { original: string; medium?: string };
  summary: string;
  season: number;
  imageUrl?: string;
}

export interface Season {
  id: number;
  url: string;
  number: number;
  name: string;
  episodeOrder: number;
  premiereDate: string | null;
  endDate: string | null;
  network: {
    id: number;
    name: string;
    country: {
      name: string;
      code: string;
      timezone: string;
    };
  };
  webChannel: any;
  image: {
    original: string | null;
  } | null;
  summary: string | null;
}

export interface Show {
  id: number;
  name: string;
  image: { original: string };
  summary: string;
}

export interface EpisodeState {
  details: Episode;
  loading: boolean;
  error: string | null;
}

export interface ShowState {
  details: Show;
  episodes: Episode[];
  seasons: Season[];
  loading: boolean;
}

export interface SeasonWithEpisodes extends Season {
  episodes: Episode[];
}
