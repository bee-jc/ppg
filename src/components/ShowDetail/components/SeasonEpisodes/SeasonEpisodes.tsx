import React from 'react';

import { Episode, SeasonWithEpisodes } from '../../../../types/types';
import EpisodeCard from '../EpisodeCard/EpisodeCard';
import { SeasonHeader, EpisodesList } from '../../../../styles';

const SeasonEpisodes: React.FC<{ season: SeasonWithEpisodes }> = ({
  season,
}) => (
  <>
    <SeasonHeader>Season {season.number}</SeasonHeader>
    <EpisodesList>
      {season.episodes.map((episode: Episode) => (
        <EpisodeCard key={episode.id} episode={episode} />
      ))}
    </EpisodesList>
  </>
);

export default SeasonEpisodes;
