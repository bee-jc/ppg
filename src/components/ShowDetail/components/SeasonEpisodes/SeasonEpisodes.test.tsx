import React from 'react';
import { render, screen } from '@testing-library/react';

import SeasonEpisodes from './SeasonEpisodes';
import { SeasonWithEpisodes, Episode } from '../../../../types/types';

jest.mock('../EpisodeCard/EpisodeCard', () => (props: any) => (
  <div data-testid={`episode-card-${props.episode.id}`}>
    {props.episode.name}
  </div>
));

describe('SeasonEpisodes', () => {
  const mockEpisodes: Episode[] = [
    {
      id: 1,
      name: 'Episode 1',
      image: {
        original: 'http://example.com/image1.jpg',
        medium: 'http://example.com/medium1.jpg',
      },
      summary: 'Summary for Episode 1',
      season: 1,
    },
    {
      id: 2,
      name: 'Episode 2',
      image: {
        original: 'http://example.com/image2.jpg',
        medium: 'http://example.com/medium2.jpg',
      },
      summary: 'Summary for Episode 2',
      season: 1,
    },
  ];

  const mockSeason: SeasonWithEpisodes = {
    id: 1,
    number: 1,
    name: 'Season 1',
    episodeOrder: 2,
    premiereDate: '2023-01-01',
    endDate: '2023-01-02',
    network: {
      id: 1,
      name: 'Network',
      country: {
        name: 'Country',
        code: 'CT',
        timezone: 'Timezone',
      },
    },
    webChannel: null,
    image: { original: 'http://example.com/season.jpg' },
    summary: 'Summary for Season 1',
    episodes: mockEpisodes,
    url: 'http://example.com/season-url', // Added url property
  };

  it('should render season header and episode cards', () => {
    const { asFragment } = render(<SeasonEpisodes season={mockSeason} />);

    // Check if season header is rendered
    expect(screen.getByText('Season 1')).toBeInTheDocument();

    // Check if episode cards are rendered
    mockEpisodes.forEach((episode) => {
      expect(
        screen.getByTestId(`episode-card-${episode.id}`)
      ).toBeInTheDocument();
      expect(screen.getByText(episode.name)).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
