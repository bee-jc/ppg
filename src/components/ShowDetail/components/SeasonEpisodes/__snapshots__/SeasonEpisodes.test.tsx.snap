// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeasonEpisodes should render season header and episode cards 1`] = `
<DocumentFragment>
  <div
    class="css-i444qj"
  >
    Season 1
  </div>
  <div
    class="css-1fagh72"
  >
    <div
      data-testid="episode-card-1"
    >
      Episode 1
    </div>
    <div
      data-testid="episode-card-2"
    >
      Episode 2
    </div>
  </div>
</DocumentFragment>
`;
